### ISSUE TITLE
[Feature request] Add all missions sent text even if last mission got skipped

### ISSUE THREAD

**Issue by zeenk** [TLDRMissions/TLDRMissions#232]
**TLDR Version of your idea**
TLDR version: Pretty much the title.

**Slightly longer version**
Right now if the last mission in the queue takes too long and gets skipped it just turns to a blank screen and doesn't say the "All missions sent" message so the sound from the WA you made does not get triggered.

It just looks like this, I turned it down to 500 just to show an example.
![2022-05-20_07-22-17](https://user-images.githubusercontent.com/68170439/169456188-c483ae6f-2202-4920-a0e8-fc923f3142ee.png)

Also a message under "All missions sent" saying if any missions got skipped would be nice :P

**Comment by teelolws** [comment]
The blank screen is a bug. I need to change the code around line 591 in guihandlers.lua to report something like "Mission skipped, no missions left".

Per https://github.com/TLDRMissions/TLDRMissions/issues/132#issuecomment-1128328439 its on my list to split that "All missions sent" message up into two types. All missions sent vs some missions sent. And with different WA events for each.

**Comment by zeenk** [comment]
Ah okay, nice 👍

**Comment by teelolws** [comment]
![image](https://user-images.githubusercontent.com/49792789/169497982-39dfb310-69dd-406e-8a0f-2138545f0077.png)

**Comment by zeenk** [comment]
Awesome! This addon just keeps getting better and better, thx 🙂

**Comment by zeenk** [comment]
Was this added to the 061c beta release or is it not ready for testing yet?

**Comment by teelolws** [comment]
No, I was just showing you what its probably going to look like. I haven't actually finished it yet though.

**Comment by zeenk** [comment]
Ah okay ^^

**Comment by teelolws** [comment]
In next patch.

**Comment by zeenk** [comment]
![2022-05-21_05-09-58](https://user-images.githubusercontent.com/68170439/169633086-728f1921-4889-4e3a-9216-10be9950034e.png)
Works great! :) 1 thing tho, seem to have broken the weakaura you made before, is there a new command to use now to trigger for this new text?

**Comment by zeenk** [comment]
Nvm I figured it out! added another trigger for TLDRMISSIONS_SENT_PARTIAL

**Comment by zeenk** [comment]
Another bug? I found when it was calculating the last mission in the queue it did not give the missions sent message at the end, got stuck on this.

![2022-05-21_06-01-05](https://user-images.githubusercontent.com/68170439/169634617-e16ee537-cbf9-4114-bead-96da61c21821.png)

**Comment by teelolws** [comment]
Nah that message is based on the last simulation failing. I can look at splitting up that message into _partial and _failed.

**Comment by teelolws** [comment]
Pushed an 062b beta into the developer repository to try. Tomorrow, I presume. I don't want to do more than one upload a day to Curseforge unless theres an Lua error going.

**Comment by zeenk** [comment]
Awesome, I'll try it tomorrow 👌

**Comment by zeenk** [comment]
Seem to be working as it should now with the summary at the end except for this one I got, the addon got stuck on this mission but it got sent out as you can see but it didn't show the missions has been sent message after 🤔 

![2022-05-21_19-42-45](https://user-images.githubusercontent.com/68170439/169663239-320b601a-0add-4329-9cb0-b7b0cf474bf5.png)

**Comment by zeenk** [comment]
Also, what does this mean?
![2022-05-21_20-26-51](https://user-images.githubusercontent.com/68170439/169664649-c42ddc4d-9708-4f01-8d75-3b9a227ca27e.png)

I was able to beat it with VenturePlan
![2022-05-21_20-27-29](https://user-images.githubusercontent.com/68170439/169664664-b03d101c-9c35-4aba-a2f9-3c2558f99a6e.png)

**Comment by zeenk** [comment]
Didn't quite work out, ran into this now, only gives this and no summary at the end.
![2022-05-21_20-55-36](https://user-images.githubusercontent.com/68170439/169665572-edb450c8-0986-48fd-a060-1b00e6736864.png)

**Comment by teelolws** [comment]
For the first image: that happens if Blizzard never sent back a MISSION_SENT_SUCCESS event. Not much I can do about Blizzard bugs!

For the third image: the last mission in its list got skipped. I'll split the error up into _PARTIAL in the next update.

**Comment by zeenk** [comment]
Sorry, more problems :P gave a blank screen now but I heard the katching sound from the WA, some missions got skipped.

![2022-05-22_04-17-51](https://user-images.githubusercontent.com/68170439/169675467-28993a9b-f017-4f5b-92e8-e8fccd78776a.png)

**Comment by zeenk** [comment]
Another weird report, shouldn't it just say All missions sent if no missions got skipped? 🤔 
![2022-05-22_04-21-45](https://user-images.githubusercontent.com/68170439/169675536-5cb8f7e0-a8e2-4595-97c8-80afbd847291.png)

**Comment by zeenk** [comment]
It does that too so dno why the other one says 0 skipped, sorry for the spam :P

![2022-05-22_04-24-59](https://user-images.githubusercontent.com/68170439/169675622-14c4d547-d9b3-487e-8ec4-9dcf40ca5214.png)

**Comment by teelolws** [comment]
For your second image: its saying that to let you know you still have 4 followers that are sitting on their lazy butts! Not sure about the other two.

**Comment by zeenk** [comment]
Oh I see ^^

**Comment by teelolws** [comment]
Actually looked back over the code. The middle image shows that because the last mission failed. It wasn't skipped, it couldn't find a way to beat the mission. There being 4 available followers is just a consequence of that.

**Comment by zeenk** [comment]
Ah but shouldn't it count as a skip then if it failed?

**Comment by teelolws** [comment]
No, "skips" means it never finished calculating it. "Fails" means it finished calculating and couldn't find a way to beat it.

Skips means you could click that Calculate button again and it'll pick up where it left off and try another 5000 simulations to see if it can find a way to win. But if it "Failed" it doesn't matter what it will do, it won't find a way to win without the minion healing more or settings being changed.

**Comment by zeenk** [comment]
I see 👍

**Comment by zeenk** [comment]
The first pic is a bit weird tho, has only happened 1 time so far.

**Comment by zeenk** [comment]
Also I see you just released 063 15mins ago, should I stick with 062c tonight or update to 063? im about half way done with my missions for tonight.

**Comment by teelolws** [comment]
Update if you can, it has some of the engine fixes you reported.

**Comment by zeenk** [comment]
Alright, updating now.

**Comment by teelolws** [comment]
With that blank screen, its actually supposed to show a message "Mission was skipped". Some loose code must be blanking out that error.

**Comment by teelolws** [comment]
Okay think I got it. If theres only 1 mission in the queue and you click the Skip button, it'll show: 
![image](https://user-images.githubusercontent.com/49792789/169675991-b4a16c52-b276-4ca8-91f9-8ac4af388c4c.png)


But if theres > 1 mission and you Skip them all, it'll show:
![image](https://user-images.githubusercontent.com/49792789/169676002-9552d76e-cfc8-47cf-bdc0-4bd990b79066.png)

**Comment by teelolws** [comment]
Beta release 063b going up on the dev branch. Its only one line of code changed to sort that blank screen out.

**Comment by zeenk** [comment]
Updated, i'll report back if I get any more weird bugs, got about 25-30 chars left to test on

**Comment by zeenk** [comment]
Done now with the remaining alts for tonight and didn't get any more weird summary issues 👍

**Comment by teelolws** [comment]
I'll work in "failed missions" into that report message. A job for tomorrow.

**Comment by teelolws** [comment]
I found another situation where it can get stuck on this:
![image](https://user-images.githubusercontent.com/49792789/169721410-94a6de64-d4b1-48be-8a85-524b85204141.png)

My character only has 11 anima, but because of the 3 troops it actually costs 13 to send that one out. It will be difficult to fix.

**Comment by zeenk** [comment]
Maybe just add a "Not enough anima to start mission" in red text above the start mission button or something? or is it something else that is hard to fix with it?

**Comment by teelolws** [comment]
The problem was working out the true cost of the mission. As of the time of writing, https://wowpedia.fandom.com/wiki/World_of_Warcraft_API#Warlords_of_Draenor doesn't have any documentation for     `C_Garrison.GetMissionCost(missionID)`. So I didn't know how to find it out.

I did some testing and found it actually returns two values: the original cost (10 anima), and the cost adjusted for troops (13 anima). And I can just work with the second return.

**Comment by zeenk** [comment]
Ah, I hope you can figure it out. That would be nice to have actually that it shows the "true cost" of the mission when it's sending them out, I had my anima cost limiter set at 160 because I didn't know if the addon was thinking about the extra cost after extra troops has been added :p

**Comment by teelolws** [comment]
The filter is applied before troops are added, so it will continue as it has so far: it filters on the base cost of the mission.

All I changed in todays patch was added the error message if you don't have enough anima to cover the base cost + troops cost.

**Comment by zeenk** [comment]
Alright ^^

**Comment by zeenk** [comment]
1 last thing I would suggest if it's possible to make a downwards list of the mission summary since the 1 line is starting to get a bit long sometimes :P if it's not too much work for u to change it ofc, just a small suggestion ^^

Current:
![2022-05-23_04-23-44](https://user-images.githubusercontent.com/68170439/169732182-10ac5f7a-18e9-4d21-87b3-6ca0bbbb6ebe.png)

Suggestion:
![169675467-28993a9b-f017-4f5b-92e8-e8fccd78776a](https://user-images.githubusercontent.com/68170439/169732148-33ab142f-aade-4781-b1db-7e0cc5033273.png)

And if all missions got sent out with no skips or fails etc just print the "All missions sent!" message like it is now ^^

**Comment by teelolws** [comment]
Its... supposed to word wrap. I'll check it out tomorrow.

**Comment by zeenk** [comment]
Did you try to do any word wrapping changes in 065? it's still sticking out on the sides, but in other news.. didn't get any failed or discrepancy to report tonight 😄  

![2022-05-24_06-32-57](https://user-images.githubusercontent.com/68170439/169949567-5cdc1f2d-55ff-44a7-b1cd-f32d1c504df3.png)

**Comment by teelolws** [comment]
I thought I did... wtf https://github.com/TLDRMissions/main/commit/df5214660b58858ba0f981624b07691a037a0033

**Comment by teelolws** [comment]
Oh. The message is going into NextMissionLabel instead of FailedCalcLabel. Whoops. May as well change the text colour while I'm messing with that.
![image](https://user-images.githubusercontent.com/49792789/169954475-54250264-4047-4a8e-bacc-d5c3f17b7092.png)

**Comment by zeenk** [comment]
Looks good 👍

**Comment by zeenk** [comment]
Done some testing now with the 065b patch and couldn't find any more weird bugs or anything, everything looks good so you can probably close this issue now unless there's something else you plan to add to it.
